python-main.js

```diff
@@ -1426,6 +1443,7 @@ function web_editor(config) {
     setupEditor(qs, migration);
     setupButtons();
     TRANSLATIONS.translateEmbedStrings(config.translate);
+    TRANSLATIONS.setLang(qs);
```

Comment:
At this point `translateEmbedStrings` is called twice, once above this line, and another time inside `setLang` (which calls `updateLang`, and `updateLang`  calls `translateEmbedStrings`.

Is there anyway we can combine these two functions into a single one that does the following:
- Check for language query parameters
- Update the js file and (with a callback) update the config object with the different language
- Updates all the embedded strings

Reply:
@microbit-carlos I've been changing the code to see how the different functions work, and can't work out the use of the translateEmbedStrings function. The website seems to load perfectly without the line: TRANSLATIONS.translateEmbedStrings(config.translate). I can add else clauses that can call updateLang even if there isn't a 'message.l' or if 'message.l' isn't in 'validLangs' but not sure if thats necessary seeing as it works without that.

Reply:
It loads without the line because it was called by updateLang, which in turn was called by setLang. So that function was called twice during start-up.
I've updated the code a bit in 5994349def1c8d930b9644f7b4fe2297740b8422 to merge the two functions, let me know if you are happy with those changes.

Reply:
OK, I thought so. I was just confused because the function call of translateEmbedStrings had the parameter config.translate, while updateLang called it with 'es' or 'pl' so I couldn't work out what the original point of having TRANSLATIONS.translateEmbedStrings(config.translate) was.